### PR TITLE
MINIFICPP-1600 Unify temporary directory creation in tests

### PR DESCRIPTION
**Pull request by fgerlits**
https://issues.apache.org/jira/browse/MINIFICPP-1600

Remove all variants of `createTempDir`/`createTempDirectory` except one, and use that one in every test.

---

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/sftp/tests/FetchSFTPTests.cpp [hunk 1]
```diff
@@ -75,10 +73,10 @@ class FetchSFTPTestsFixture {
...
 
     // Create temporary directories
-    testController.createTempDirectory(src_dir);
-    REQUIRE(src_dir != nullptr);
-    testController.createTempDirectory(dst_dir);
-    REQUIRE(dst_dir != nullptr);
+    src_dir = testController.createTempDirectory();
+    REQUIRE_FALSE(src_dir.empty());
+    dst_dir = testController.createTempDirectory();
+    REQUIRE_FALSE(dst_dir.empty());
```
Comment (szaszm) [inline extensions/sftp/tests/FetchSFTPTests.cpp:79]: Did you consider moving the initializers to the member initializer list or to default member initializers?

File: encrypt-config/tests/CMakeLists.txt [hunk 2]
```diff
@@ -32,7 +32,7 @@ foreach(testfile ${ENCRYPT_CONFIG_TESTS})
   target_include_directories(${testfilename} BEFORE PRIVATE "${CMAKE_SOURCE_DIR}/libminifi/test")
 
   target_wholearchive_library(${testfilename} minifi)
-  target_link_libraries(${testfilename} ${CATCH_MAIN_LIB})
+  target_link_libraries(${testfilename} ${CATCH_MAIN_LIB} ${TEST_BASE_LIB})
```
Comment (szaszm) [inline encrypt-config/tests/CMakeLists.txt:35]: Is this a new library, or why is this addition necessary?

**Review by szaszm — COMMENTED** [review]
Thanks for your effort! Just a few small questions.

File: encrypt-config/tests/CMakeLists.txt (hunk 2 above)
Comment (fgerlits) [reply encrypt-config/tests/CMakeLists.txt:35]: It's needed because `encrypt-config/tests/ConfigFileTests.cpp` now uses `TestController`, which is in `test_base` (= `${TEST_BASE_LIB}`.  Before this change, `ConfigFileTests` created the temporary directory directly using `utils::file::create_temp_directory()` -- which also means that this test directory was not cleaned up after the test.

File: extensions/sftp/tests/FetchSFTPTests.cpp (hunk 1 above)
Comment (fgerlits) [reply extensions/sftp/tests/FetchSFTPTests.cpp]: I did not consider it, but it is a good idea, thanks!  Done in https://github.com/apache/nifi-minifi-cpp/pull/1139/commits/555d7b367f11d7750d594f2c0076b1e4c4144cdd.

File: encrypt-config/tests/CMakeLists.txt (hunk 2 above)
Comment (szaszm) [reply encrypt-config/tests/CMakeLists.txt:35]: I see, thanks.

**Review by szaszm — APPROVED** [review]

File: extensions/http-curl/tests/unit/InvokeHTTPTests.cpp [hunk 3]
```diff
@@ -47,8 +47,7 @@ TEST_CASE("HTTPTestsWithNoResourceClaimPOST", "[httptest1]") {
 
   std::shared_ptr<core::Processor> logAttribute = std::make_shared<org::apache::nifi::minifi::processors::LogAttribute>("logattribute");
 
-  char format[] = "/tmp/gt.XXXXXX";
-  auto dir = testController.createTempDirectory(format);
+  auto dir = testController.createTempDirectory();
```
Comment (aminadinari19) [inline extensions/http-curl/tests/unit/InvokeHTTPTests.cpp:50]: Maybe I am wrong but the dir variable doesn't seem to be used anywhere in this test.

File: extensions/http-curl/tests/unit/InvokeHTTPTests.cpp [hunk 4]
```diff
@@ -168,8 +167,7 @@ TEST_CASE("HTTPTestsWithResourceClaimPOST", "[httptest1]") {
...
```
Comment (aminadinari19) [inline extensions/http-curl/tests/unit/InvokeHTTPTests.cpp:170]: Same concern as the above.

File: extensions/standard-processors/tests/unit/RetryFlowFileTests.cpp [hunk 5]
```diff
@@ -86,7 +85,7 @@ class RetryFlowFileTest {
       optional<bool> processor_uuid_matches_flowfile) {
     reInitialize();
 
-    const std::string output_dir = createTempDir(testController_.get());
+    const std::string output_dir = testController_->createTempDirectory();
```
Comment (lordgamez) [inline extensions/standard-processors/tests/unit/RetryFlowFileTests.cpp:88]: This `output_dir` seems to be unused

File: extensions/http-curl/tests/unit/InvokeHTTPTests.cpp (hunk 3 above)
Comment (fgerlits) [reply extensions/http-curl/tests/unit/InvokeHTTPTests.cpp]: Good catch, thanks!  I have removed the unused variables in https://github.com/apache/nifi-minifi-cpp/pull/1139/commits/d61c4e651a4e3b1c36db8da6f8352ec65e76505a.

File: extensions/standard-processors/tests/unit/RetryFlowFileTests.cpp (hunk 5 above)
Comment (fgerlits) [reply extensions/standard-processors/tests/unit/RetryFlowFileTests.cpp]: Yes, it is.  Removed it in https://github.com/apache/nifi-minifi-cpp/pull/1139/commits/4aae1742792f939879a50a211fb39877c068925a.

File: extensions/sftp/tests/FetchSFTPTests.cpp [hunk 6]
```diff
@@ -74,18 +72,15 @@ class FetchSFTPTestsFixture {
...
     LogTestController::getInstance().setDebug<SFTPTestServer>();
 
-    // Create temporary directories
-    testController.createTempDirectory(src_dir);
-    REQUIRE(src_dir != nullptr);
-    testController.createTempDirectory(dst_dir);
-    REQUIRE(dst_dir != nullptr);
+    REQUIRE_FALSE(src_dir.empty());
+    REQUIRE_FALSE(dst_dir.empty());
+    REQUIRE(plan);
```
Comment (aminadinari19) [inline extensions/sftp/tests/FetchSFTPTests.cpp:77]: What is the purpose of the last REQUIRE statement?

**Review by lordgamez — APPROVED** [review]

File: extensions/sftp/tests/FetchSFTPTests.cpp (hunk 6 above)
Comment (fgerlits) [reply extensions/sftp/tests/FetchSFTPTests.cpp:77]: To make sure that the pointer is not null, since we dereference it later.  An assertion failure here is better than a segmentation fault later.

Comment (aminadinari19) [reply extensions/sftp/tests/FetchSFTPTests.cpp:77]: Got it. Thanks :)

File: libminifi/test/bustache-tests/ApplyTemplateTests.cpp [hunk 7]
```diff
@@ -85,13 +85,13 @@ TEST_CASE("Test usage of ApplyTemplate", "[ApplyTemplateTest]") {
     std::shared_ptr<core::Processor> atprocessor = plan->addProcessor("ApplyTemplate", "testApplyTemplate", core::Relationship("success", "description"), true);
 
     std::shared_ptr<core::Processor> putfile = plan->addProcessor("PutFile", "putfile", core::Relationship("success", "description"), true);
-    plan->setProperty(putfile, org::apache::nifi::minifi::processors::PutFile::Directory.getName(), dir3);
+    plan->setProperty(putfile, org::apache::nifi::minifi::processors::PutFile::Directory.getName(), put_file_destination_dir);
     plan->setProperty(putfile, org::apache::nifi::minifi::processors::PutFile::ConflictResolution.getName(),
```
Comment (aminadinari19) [inline libminifi/test/bustache-tests/ApplyTemplateTests.cpp:89]: I like how the variables are way more meaningful now 👍

**Review by aminadinari19 — APPROVED** [review]
